sqlmesh/core/plan/builder.py

```diff
@@ -553,6 +583,22 @@ def _set_choice(
 
             is_forward_only_child = self._is_forward_only_change(child_s_id)
 
+            if is_forward_only_child:
+                # no DAG if manual categorization
+                if not dag:
+                    logger.info(
+                        f"Unable to evaluate model '{child_s_id.name}'s upstream parents for destructive schema changes at plan time."
+                    )
+
+                self._validate_destructive_change(
+                    child_s_id,
+                    (
+                        self._upstream_directly_modified(child_s_id, directly_modified, dag)
```

Comment:
_upstream_directly_modified is called 3x with the same argument, always with validate_destructive_change.

can we move this call to validate_destructive_change since it seems like it's required?